src/jit/importer.cpp

```diff
@@ -13165,56 +13193,50 @@ MATH_MAYBE_CALL_NO_OVF:         ovfl = false;
 
                 assert(helper == CORINFO_HELP_UNBOX_NULLABLE && "Make sure the helper is nullable!");
 
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-                if (varTypeIsStruct(op1))
+#if FEATURE_MULTIREG_RET
+
+                if (varTypeIsStruct(op1) && IsMultiRegReturnedType(resolvedToken.hClass))
                 {
-                    if (IsMultiRegReturnedType(resolvedToken.hClass))
-                    {
-                        // Unbox nullable helper returns a TYP_STRUCT.
-                        // We need to spill it to a temp so than we can take the address of it.
-                        // We need the temp so we can pass its address to the unbox_nullable jit helper function.
-                        // This is needed for nullables returned in 2 registers.
-                        // The ones returned in a single register are normalized.
-                        // For the bigger than 16 bytes nullables there is retbuf already passed in 
-                        // rdi/rsi (depending whether there is a "this").
-
-                        unsigned   tmp = lvaGrabTemp(true DEBUGARG("UNBOXing a register returnable nullable"));
-                        lvaTable[tmp].lvIsMultiRegArg = true;
-                        lvaSetStruct(tmp, resolvedToken.hClass, true  /* unsafe value cls check */);
-
-                        op2 = gtNewLclvNode(tmp, TYP_STRUCT);
-                        op1 = impAssignStruct(op2, op1, resolvedToken.hClass, (unsigned)CHECK_SPILL_ALL);
-                        assert(op1->gtType == TYP_VOID); // We must be assigning the return struct to the temp.
-
-                        op2 = gtNewLclvNode(tmp, TYP_STRUCT);
-                        op2 = gtNewOperNode(GT_ADDR, TYP_BYREF, op2);
-                        op1 = gtNewOperNode(GT_COMMA, TYP_BYREF, op1, op2);
-
-                        // In this case the return value of the unbox helper is TYP_BYREF.
-                        // Make sure the right type is placed on the operand type stack.
-                        impPushOnStack(op1, tiRetVal);
+                    // Unbox nullable helper returns a TYP_STRUCT.
+                    // We need to spill it to a temp so than we can take the address of it.
+                    // We need the temp so we can pass its address to the unbox_nullable jit helper function.
+                    // This is needed for nullables returned yuusing multiple registers.
```

Comment:
Typo: "using". Also, the above is a bit repetitive, I might say "We need to spill it to a temp so that we can pass the address to the unbox_nullable jit helper function".